README.md
# Cookie Service

[![npm version](https://badge.fury.io/js/cookie-service-banner.svg)](https://badge.fury.io/js/cookie-service-banner)
[![Build Status](https://travis-ci.org/liyokuna/cookie-consent.svg?branch=master)](https://travis-ci.org/liyokuna/cookie-consent)
[![dependencies Status](https://david-dm.org/liyokuna/cookie-consent/status.svg)](https://david-dm.org/liyokuna/cookie-consent)

Cookie Service focuses on accessibility and on respecting the [GDPR](https://fr.wikipedia.org/wiki/R%C3%A8glement_g%C3%A9n%C3%A9ral_sur_la_protection_des_donn%C3%A9es).
If you are using [Google Analytics](https://analytics.google.com/analytics/web/), this will suit to your project because this library focus on disable Google Ananlytics when the user decide to do so.
This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 7.3.8.

## Demo
Here is a link to the [demo](https://cookie-conset-banner.netlify.com/).

## Dependencies

[Angular](https://angular.io/) ( the latest version of Angular 7+ )
[Bootstrap](https://getbootstrap.com/) ( the latest version of Bootsrap 4+ )

## Installation

Install cookie-service dependency through npm.
`npm install cookie-service-banner`

After installing, you need to import the main module:
`import { CookieServiceModule } from 'cookie-service-banner';`

The full configuration:

`
import { CookieServiceModule, CookieConfig } from 'cookie-service-banner';<br>

const testLibConfig: CookieConfig = {
  header: {
    title:"Cookie Consent Banner",
    message: "This website uses cookie to provide your the best experience. ",
    domain:"localhost",
    ga_id: "UA-123456-1",
    color: '#fff',
    bcolor: '#000'
  },
  acceptButton: {
    enable: false,
    accept: "Got it!",
    color: '#fff',
    bcolor: '#266433'
  },
  allowButton: {
    enable: true,
    allow: "Allow Cookie",
    color: '#000',
    bcolor: '#f36e15f5'
  },
  declineButton: {
    enable: true,
    deny: "Refuse Cookie",
    color: '#000',
    bcolor: '#fff'
  },
  learnMoreLink: {
    enable: true,
    learnMore: "learn more",
    link: "www.example.com",
    color: '#3D9BFF'
  },
  review: {
    enable: true,
    message: "Review My consentement",
    color: "",
    bcolor: "",
  }
}

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    NgbModule,
    CookieServiceModule,
    CookieServiceModule.forRoot(testLibConfig),
    AppRoutingModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
`

## Usage

Once the steps mentioned above are done, you can use import library selector in your app.component.html.

`
...
<router-outlet></router-outlet>
<lib-cookie-service (isOpened)="onOpen($event)" ></lib-cookie-service>
`

`isOpened` is function meant to state the openning or the closing of the cookie banner.

## Development server

Run `ng serve` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.

## Build

Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.

## Running unit tests

Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.io).

## Running end-to-end tests

Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).

## Further help

To get more help on the Angular CLI use `ng help` or use the [Angular CLI README](https://github.com/angular/angular-cli/blob/master/README.md).

#License

Copyright(c) 2019 Lionel Mukuna Ciowela.


src/CHANGELOG.md
<a name="2.2.0"></a> (26/04/2020)
## 2.2.0

### Changes:
*	**Travis CI:** Updating version o f node js engine to run test and deploy [302b7e](https://github.com/liyokuna/cookie-consent/commit/302b7e116c96b0e130e66557befd9922b2a2412e)
*	**Package JSON :** Updating angular to 9 [7d7798](https://github.com/liyokuna/cookie-consent/commit/7d77985faca17a0d18231a767fecc54988f5d568)
*	**Cookie Service Banner :** correcting a condition to reject cookie for google manager tracking [9ad42e](https://github.com/liyokuna/cookie-consent/commit/9ad42e009bbf9bb3b5f2976d1563fecd779398ce)
*	**Example project :** Adding google tag manager [9eb257](https://github.com/liyokuna/cookie-consent/commit/9eb2574239bf36f583e45f877591cbbff0551307)
*	**Example project :** Correcting the name of the cookie into the file example [1e7776](https://github.com/liyokuna/cookie-consent/commit/1e776dab3e352ed4d788cfb2d40ebe67dfc69cd6)

<a name="2.0.1"></a> (15/11/2019)
## 2.0.1

### Correction:
*	**Cookie-service:** Removing domain from the saving cookie schema [35ce3c7](https://github.com/liyokuna/cookie-consent/commit/35ce3c7b32e2a321f68851ba45762423f01541c1)

<a name="1.0.5"></a> (08/08/2019)
## 2.0.0

### Changes:
*	**Travis CI:** Adding auto deployement to npm and github pages [fea856e](https://github.com/liyokuna/cookie-consent/commit/fea856efcdfe24c3d11990c3d2dd1af768b2d660)
*	**Package JSON :** Updating angular to 8 [ebf317](https://github.com/liyokuna/cookie-consent/commit/ebf3175224d3951ab2e91b17d8124190ccdb8fae)

<a name="1.0.5"></a> (08/07/2019)
## 1.0.5

### Changes:
*	**Cookie-Service :** Improvement done on testing the component [7e18d8](https://github.com/liyokuna/cookie-consent/commit/7e18da873412729137fb90dac7c9ab92327376e2)
*	**Cookie-Service :** Adding title attributes on button, a, div html tags [d3d4517](https://github.com/liyokuna/cookie-consent/commit/d3d4517c68f43c913a73c68200140fc03e3f419d)

src/app/app.module.ts
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';

import { CookieServiceModule, CookieConfig } from 'cookie-service-banner';

const testLibConfig: CookieConfig = {
  header: {
    title:'Cookie Consent Banner',
    message: 'This website uses cookies to provide you the best experience. ',
    domain:'localhost',
    ga_id: 'UA-80915848-2',
    color: '#fff',
    bcolor: '#000'
  },
  acceptButton: {
    enable: false,
    accept: 'Got it!',
    color: '#fff',
    bcolor: '#266433'
  },
  allowButton: {
    enable: true,
    allow: 'Allow Cookie',
    color: '#000',
    bcolor: '#f36e15f5'
  },
  declineButton: {
    enable: true,
    deny: 'Refuse Cookie',
    color: '#000',
    bcolor: '#fff'
  },
  learnMoreLink: {
    enable: true,
    learnMore: 'learn more',
    link: 'www.example.com',
    color: '#3D9BFF'
  },
  review: {
    enable: true,
    message: 'Review My consentement',
    color: '',
    bcolor: '',
  }
}

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    CookieServiceModule,
    CookieServiceModule.forRoot(testLibConfig),
    AppRoutingModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
